datadog_checks_base/datadog_checks/base/stubs/aggregator.py

```diff
@@ -207,20 +212,22 @@ def assert_histogram_bucket(
             condition=condition, msg=msg, expected_stub=expected_bucket, submitted_elements=self._histogram_buckets
         )
 
-    def assert_metric(self, name, value=None, tags=None, count=None, at_least=1, hostname=None, metric_type=None):
+    def assert_metric(
+        self, name, value=None, tags=None, count=None, at_least=1, hostname=None, metric_type=None, device=None
+    ):
```

Comment:
I'm not sure, we should try to add device to the aggregator (we can just not assert device in e2e).

We are using the `aggregator` for assertions, it make sense for integrations tests (submission metrics).

We are also using `aggregator` for e2e assertions, but we are already stretching its scope by using it for e2e assertions. By adding the device to it, we would stretch its scope even more, and I think we should avoid that.